### ISSUE TITLE
PE resources parsed incorrectly

### ISSUE THREAD

**Issue by williballenthin** [lief-project/LIEF#245]
I've noticed that LIEF occasionally parses PE resources incorrectly; specifically, string items from the version info.

As an example, consider `71b201a5a7dfdbe91c0a7783f845b71d066c62014b944f488de5aec6272f907c`, which is available via VT (let me know if I should send this directly).

Resource Hacker parses the following structures:

![image](https://user-images.githubusercontent.com/156560/50116160-a0a8ff00-0206-11e9-8360-b2b98ecd506e.png)

Note that there are values like: `CompanyName`=`Microsoft Corporation. All rights reserved.`

However, LIEF parses values differently:

![image](https://user-images.githubusercontent.com/156560/50116221-ce8e4380-0206-11e9-897d-924a24000897.png)

```
'.7600.16385': ' ',
'1, 7600, 16385': ' ',
'Comments': 'x,\x01CompanyName',
'ecialBuild': 'D',
'galTrademarks': '(',
'iginalFilename': ' ',
'ivateBuild': ' ',
'leDescription': '>\x0f\x01FileVersion',
'oductName': 'H\x12\x01ProductVersion',
'rosoft Corporation. All rights reserved.': '(',
'ternalName': 'J\x13\x01LegalCopyright',
'yright (C) 2014': '('
```

I will begin to look into the parsing; however, I'm not yet familiar with the LIEF codebase, so cannot promise that this will be efficient.

**Comment by williballenthin** [comment]
I took some time to manually parse the version resource for this sample, and I believe it illustrates where the bug is.

![image](https://user-images.githubusercontent.com/156560/50119763-faaec200-0210-11e9-8462-339c2904d21a.png)

Starting after the `StringFileInfo` header, there is the `key` field (offset 0x80), and then the list of key-value pairs. The illustration above shows two key-value pairs parsed out: in yellow (offset 0x98), and in blue (0xB0).

I think LIEF may incorrectly parse the yellow key-value pair, probably because the value portion is unexpectedly empty. This can be determined in two ways:
  1. The `string length in words` field is 0x0, and 
  2. the current position after the key ("Comment") equals the total element size (0x18).

I think the bug can be fixed by adding additional checks around line [ResourceManager.cpp::605](https://github.com/lief-project/LIEF/blob/1f32b253295535de77f81997ac99c49fb8015c83/src/PE/ResourcesManager.cpp#L605) to:
  1. assert `string_value_length` is greater than zero, and
  2. assert `stream.pos() < element_start+ string_length` (`$element_start$` should be initialized to `stream.pos()` before line 592).
  3. (optional) rename `string_length` to `element_length` or something similar, as it indicates the size of key and value combined.

I don't have a build environment for LIEF setup yet, but I may attempt this to submit a PR.

**Comment by romainthomas** [comment]
Resolved with #246
